llama-index-integrations/llms/llama-index-llms-deepinfra/llama_index/llms/deepinfra/base.py

```diff
@@ -0,0 +1,538 @@
+import aiohttp
+import requests
+
+from typing import (
+    Any,
+    Callable,
+    Dict,
+    Optional,
+    Sequence,
+    AsyncGenerator,
+    Generator,
+)
+from llama_index.core.bridge.pydantic import Field, PrivateAttr
+from llama_index.core.llms.llm import LLM
+from llama_index.core.constants import DEFAULT_TEMPERATURE
+
+from llama_index.core.types import BaseOutputParser, PydanticProgramMode
+from llama_index.core.callbacks import CallbackManager
+from llama_index.core.base.llms.generic_utils import get_from_param_or_env
+from llama_index.core.base.llms.types import (
+    CompletionResponse,
+    CompletionResponseGen,
+    ChatResponse,
+    ChatResponseGen,
+    ChatResponseAsyncGen,
+    CompletionResponseAsyncGen,
+    LLMMetadata,
+    MessageRole,
+    ChatMessage,
+)
+
+from llama_index.core.llms.callbacks import (
+    llm_chat_callback,
+    llm_completion_callback,
+)
+
+from llama_index.llms.deepinfra.utils import (
+    maybe_decode_sse_data,
+    chat_messages_to_list,
+    retry_request,
+    aretry_request,
+)
+
+"""DeepInfra API base URL."""
+API_BASE = "https://api.deepinfra.com/v1"
+"""DeepInfra Inference API endpoint."""
+INFERENCE_ENDPOINT = "inference"
+"""Chat API endpoint for DeepInfra."""
+CHAT_API_ENDPOINT = "openai/chat/completions"
```

Comment:
I'd use `/v1/openai/chat/completions` for chat and `/v1/openai/completions` for text. The `v1` is not part of the API_BASE (some of our apis don't have v1, and it is more of a per-path thing, not global.